packages/grafana-data/src/types/templateVars.ts

```diff
@@ -195,6 +195,12 @@ export interface BaseVariableModel {
   error: any | null;
   description: string | null;
   usedInRepeat?: boolean;
+  source?: SourceRef;
```

Comment:
We don't _necessarily_ need to extend the `TypedVariableModel`, as technically we only need to persist the difference between default and other variables _in scenes_, but having this here greatly helps to simplify the implementation (we can just enrich and pass down the `TypedVariableModel` objects) and let the transform functions to handle the rest.